### PR TITLE
ngfw-13258 default alert for Configuration Backup Failure added

### PR DESCRIPTION
**Pull request by DhirajM09**
Changes:

1. Added code to include default _Configuration Backup Failure_ event if doesn't exists already.
2. Removed code for adding default _Critical Alert Event_ event as it was part of last upgrade

Screenshot from 2024-03-12 11-28-23

Screenshot from 2024-03-12 15-46-02

### CODE REVIEW

File: uvm/impl/com/untangle/uvm/EventManagerImpl.java [hunk 1]
```diff
@@ -60,7 +60,7 @@
  */
 public class EventManagerImpl implements EventManager
 {
-    private static final Integer SETTINGS_CURRENT_VERSION = 3;
+    private static final Integer SETTINGS_CURRENT_VERSION = 4;
```
Comment (cblaise-untangle) [inline uvm/impl/com/untangle/uvm/EventManagerImpl.java:63]: EventSettings.java has a default version as well.

**Comment by cblaise-untangle** [comment]
Let's also start a test suite for "alerts" to verify we catch this condition.

We have a couple in test_uvm.py (test_050_alert_rule,test_060_customized_email_alert ), but we should do a new alert suite, move those tests into it, and then create a new test for verifying the backup alert.  The test_050_alert_rule shows how to look at the alert event table to verify.  The work for the new test will be to:

1. Install configuration backup app.
2. Add a DNS entry for boxbackup.untangle.com to an unreachable address.
3. Trigger configuration backup.
4. Verify configuration backup alert.
5. Remove DNS entry.

File: uvm/impl/com/untangle/uvm/EventManagerImpl.java (hunk 1 above)
Comment (DhirajM09) [reply uvm/impl/com/untangle/uvm/EventManagerImpl.java:63]: Updated the version in EventSettings

File: uvm/impl/com/untangle/uvm/EventManagerImpl.java [hunk 2]
```diff
@@ -436,40 +436,41 @@ private void updateSettings(EventSettings settings){
...
-
-        boolean criticalFlag = false;
-        boolean reportsFlag = false;
-
-        // search for the CriticalAlertEvent REPORTS rule
-        for (EventRule er : settings.getAlertRules()) {
-            for(EventRuleCondition c : er.getConditions()) {
-                if(c.getField().equals("class") && c.getFieldValue().equals("*CriticalAlertEvent*")){
-                    criticalFlag = true;
+            boolean confBackUpFlag = false;
```
[the remark this answers is not in the document]
Comment (DhirajM09) [reply uvm/impl/com/untangle/uvm/EventManagerImpl.java:402]: Removed the rule modification code

**Comment by DhirajM09** [comment]
> Let's also start a test suite for "alerts" to verify we catch this condition.
> 
> We have a couple in test_uvm.py (test_050_alert_rule,test_060_customized_email_alert ), but we should do a new alert suite, move those tests into it, and then create a new test for verifying the backup alert. The test_050_alert_rule shows how to look at the alert event table to verify. The work for the new test will be to:
> 
> 1. Install configuration backup app.
> 2. Add a DNS entry for boxbackup.untangle.com to an unreachable address.
> 3. Trigger configuration backup.
> 4. Verify configuration backup alert.
> 5. Remove DNS entry.

@cblaise-untangle added alert-tests suite moved existing tests to new file and added test for configuration backup failure event.

File: uvm/hier/usr/lib/python3/dist-packages/tests/test_alert.py [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+        events = global_functions.get_events('Events','Alert Events',None,10)
+        found = global_functions.check_events( events.get('list'), 5,
+                                            'description', 'Configuration backup failed' )
+        
+        uvmContext.networkManager().setNetworkSettings(orig_network_settings)
+
+        assert(events != None)
+        assert(found)
+
+test_registry.register_module("alert-tests", AlertTests)
```
Comment (cblaise-untangle) [inline uvm/hier/usr/lib/python3/dist-packages/tests/test_alert.py:157]: Lets just call it "alerts"

File: uvm/hier/usr/lib/python3/dist-packages/tests/test_alert.py [hunk 4]
```diff
@@ -0,0 +1,157 @@
+import copy
```
Comment (cblaise-untangle) [inline uvm/hier/usr/lib/python3/dist-packages/tests/test_alert.py:1]: Let's rename to test_alerts.py

File: uvm/hier/usr/lib/python3/dist-packages/tests/test_alert.py (hunk 3 above)
Comment (DhirajM09) [reply uvm/hier/usr/lib/python3/dist-packages/tests/test_alert.py]: changed to **alerts**

File: uvm/hier/usr/lib/python3/dist-packages/tests/test_alert.py (hunk 4 above)
Comment (DhirajM09) [reply uvm/hier/usr/lib/python3/dist-packages/tests/test_alert.py]: file renamed to **test_alerts.py**

**Review by cblaise-untangle — APPROVED** [review]

**Pull request merged**